components/arm/fake/fake.go

```diff
@@ -68,28 +81,7 @@ func NewArm(cfg config.Component, logger golog.Logger) (arm.LocalArm, error) {
 	var model referenceframe.Model
 	var err error
 	if cfg.ConvertedAttributes != nil {
-		converted := cfg.ConvertedAttributes.(*AttrConfig)
-
-		if converted.FailNew {
-			return nil, errors.New("whoops! failed to start up")
-		}
-
-		switch resource.ModelName(converted.ArmModel) {
-		case xarm.ModelName6DOF.Name:
-			model, err = xarm.Model(cfg.Name, 6)
-		case xarm.ModelName7DOF.Name:
-			model, err = xarm.Model(cfg.Name, 7)
-		case ur.ModelName.Name:
-			model, err = ur.Model(cfg.Name)
-		case yahboom.ModelName.Name:
-			model, err = yahboom.Model(cfg.Name)
-		case eva.ModelName.Name:
-			model, err = eva.Model(cfg.Name)
-		case ModelName.Name, "":
-			model, err = referenceframe.UnmarshalModelJSON(fakeModelJSON, cfg.Name)
-		default:
-			return nil, errors.Errorf("fake arm cannot be created, unsupported arm_model: %s", cfg.ConvertedAttributes.(*AttrConfig).ArmModel)
-		}
+		model, err = modelFromName(cfg.ConvertedAttributes.(*AttrConfig).ArmModel, cfg.Name)
 	} else {
 		model, err = referenceframe.UnmarshalModelJSON(fakeModelJSON, cfg.Name)
```

Comment:
Not sure if having this around is ok or not.  This will only be the case if we make something without ConvertedAttributes, which may or not be possible from the config

Reply:
Maybe the better thing to do is just get rid of the default model entirely and for testing purposes construct fake arms with known models (i.e. UR5)

Reply:
@spieswl @biotinker lmk

Reply:
To Ray's question, I agree. Fake arm should no longer have a "default" kinematic model, and the tests that use it should pass in that test json explicitly from some testing location.

Tangential to the question above, I think this is still missing a piece of functionality we want.
As it stands now:
- If someone wants to simulate an arm whose kinematics we support, we use `fake`, as above.
- If someone wants to replace the kinematics on an arm we support with their own kinematics because they altered it, they pass a new file into `wrapper arm` in addition to the real arm.

But, if someone wants to simulate an arbitrary kinematic chain from a file without moving anything, that currently requires `fake` with arbitrary inputs, plus a wrapper. `fake` should have the ability to take a file as well. 

This would then be used to receive what's currently the fake arm's default kinematics for testing.

Reply:
If it's not possible to `Unmarshal`, we will return an error from the `referenceframe` library, so we shouldn't be operating on something incorrect. I would prefer to at least allow us to specify non-standard kinematics models in for the fake arm for testing purposes without having to modify the `fake` component each time. My vote is to keep the `else` statement.

Reply:
I'm good with this.  And to expand on it we should use the `arm-model` field as the field to specify the path to the model file, as is done with the wrapper field `model-path`.  We maybe should also make sure we are consistent and change `model-path` to arm-model` to be consistent.

Reply:
Good with Peter's suggestion as well

Reply:
the `else` case does not get hit. Suppose you have config:
```
{
    "components": [
        {
            "name": "arm1",
            "type": "arm",
            "model": "fake"
           }]}
````
Here we define no `ModelPath` or `ArmModel`.
In `components/arm/fake/fake.go/NewArm()` we have conditional `if cfg.ConvertedAttributes != nil`.
However, if we print `cfg.ConvertedAttributes` we get `&{ }`.
Hence, `if cfg.ConvertedAttributes != nil` is always true and we never reach the `else` statement.

Reply:
the `else` case does not get hit. Suppose you have config:
```
{
    "components": [
        {
            "name": "arm1",
            "type": "arm",
            "model": "fake"
           }]}
````
Here we define no `ModelPath` or `ArmModel`.
In `components/arm/fake/fake.go/NewArm()` we have conditional `if cfg.ConvertedAttributes != nil`.
However, if we print `cfg.ConvertedAttributes` we get `&{ }`.
Hence, `if cfg.ConvertedAttributes != nil` is always true and we never reach the `else` statement.